deploy/cloudformation/manifest-pipeline.yml

```diff
@@ -303,6 +303,9 @@ Resources:
                 Rules:
                   - Name: suffix
                     Value: 'main.csv'
+      Environment:
+        Variables:
+          STATE_MACHINE: !Sub "arn:aws:states:${AWS::Region}:${AWS::AccountId}:stateMachine:${AWS::StackName}-StateMachine"
```

Comment:
Would `!Ref StateMachine` give you the ARN here? Would be a little less sensitive to naming changes if so. Looking at https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/intrinsic-function-reference-getatt.html, it only allows for getting the Name for a state machine, so I would assume !Ref would give you the ARN, but nothing would surprise me at this point.

Reply:
`!Ref` will get you the ARN per [AWS.](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/intrinsic-function-reference-ref.html)

Reply:
I 100% agree and is what I tried first.  However, using !Ref StateMachine causes a circular dependency between the lambda, permissions, and s3 bucket creation.

To avoid this AWS recommends doing what I've done.  In this article they are referencing the exact situation we are implementing.  It's a common problem :/
https://aws.amazon.com/premiumsupport/knowledge-center/unable-validate-circular-dependency-cloudformation/

Here are there other 'workarounds' but they are not as easy to implement and can come with other baggage.
 * Create the bucket without notification configuration, and then add it in the next stack update.
 * Create a less-constrained Lambda permission. For example, allow invocations for a specific AWS account by omitting the SourceArn.
 * Create a custom resource to run at the end of the stack workflow. This resource adds the notification configuration to the bucket after all other resources are created.